### PR TITLE
Feature user signup

### PR DESCRIPTION
**Pull request by MartynaAnnaGottschling**

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -18,8 +18,8 @@
 def api_root(request, format=None):
     if not request.user.is_authenticated:
         return Response({
-                'users': reverse(
-                    'users-list',
+                'registration': reverse(
```
Comment (TheCM) [inline users/views.py]: I don't see option for registration at `localhost:8000` when I am not logged in.

File: time_monkey/settings/base.py [hunk 2]
```diff
@@ -42,6 +42,14 @@
     'django_countries',
```
[the remark this answers is not in the document]
Comment (MartynaAnnaGottschling) [reply time_monkey/settings/base.py]: App allauth  is not included in django-allauth, which I added?

File: time_monkey/urls.py [hunk 3]
```diff
@@ -12,10 +12,11 @@
 """
 from django.conf.urls import url
 from django.contrib import admin
-from django.conf.urls import include
+from django.urls import include, path
```
[the remark this answers is not in the document]
Comment (MartynaAnnaGottschling) [reply time_monkey/urls.py]: Done

File: users/serializers.py [hunk 4]
```diff
@@ -66,3 +66,51 @@ class Meta:
...
+        if data['password1'] != data['password2']:
+            raise serializers.ValidationError(
+                _("The two password fields didn't match."))
+        return data
+
+    def get_cleaned_data(self):
+        return {
+            'first_name': self.validated_data.get('first_name', ''),
+            'last_name': self.validated_data.get('last_name', ''),
+            'password1': self.validated_data.get('password1', ''),
```
[the remark this answers is not in the document]
Comment (MartynaAnnaGottschling) [reply users/serializers.py]: Done

File: users/serializers.py [hunk 5]
```diff
@@ -66,3 +66,51 @@ class Meta:
...
+            if email and email_address_exists(email):
+                raise serializers.ValidationError(
+                    _("A user is already registered with this e-mail address."))
+        return email
+
+    def validate_password1(self, password):
+        return get_adapter().clean_password(password)
+
+    def validate(self, data):
+        if data['password1'] != data['password2']:
```
[the remark this answers is not in the document]
Comment (MartynaAnnaGottschling) [reply users/serializers.py]: Done

File: users/serializers.py [hunk 6]
```diff
@@ -66,3 +66,51 @@ class Meta:
...
+
+    def validate_email(self, email):
+        email = get_adapter().clean_email(email)
+        if allauth_settings.UNIQUE_EMAIL:
+            if email and email_address_exists(email):
+                raise serializers.ValidationError(
+                    _("A user is already registered with this e-mail address."))
+        return email
+
+    def validate_password1(self, password):
```
[the remark this answers is not in the document]
Comment (MartynaAnnaGottschling) [reply users/serializers.py]: Done

**Pull request closed without merging**